src/functions/asyncCredential/asyncCredentialHandler.test.ts

```diff
@@ -958,41 +961,78 @@ describe("Async Credential", () => {
+          const event = buildRequest({
+            headers: { Authorization: `Bearer ${jwtBuilder.getEncodedJwt()}` },
+            body: JSON.stringify({
+              state: "mockState",
+              sub: "mockSub",
+              client_id: "mockClientId",
+              govuk_signin_journey_id: "mockGovukSigninJourneyId",
+            }),
+          });
+          dependencies.getSessionService = () =>
+            new MockSessionServiceGetSessionBySubFailure(
+              env.SESSION_TABLE_NAME,
+              env.SESSION_TABLE_SUBJECT_IDENTIFIER_INDEX_NAME,
+            );
 
-        const result = await lambdaHandler(event, dependencies);
+          const result = await lambdaHandler(event, dependencies);
 
-        expect(result).toStrictEqual({
-          headers: { "Content-Type": "application/json" },
-          statusCode: 500,
-          body: JSON.stringify({
-            error: "server_error",
-            error_description: "Server Error",
-          }),
+          expect(result).toStrictEqual({
+            headers: { "Content-Type": "application/json" },
+            statusCode: 500,
+            body: JSON.stringify({
+              error: "server_error",
+              error_description: "Server Error",
+            }),
+          });
+        });
+      });
+
+      describe("Given service returns success response", () => {
+        describe("Given response value is the sessionId string", () => {
+          it("Returns 200 session recovered response", async () => {
+            const jwtBuilder = new MockJWTBuilder();
+            const event = buildRequest({
+              headers: {
+                Authorization: `Bearer ${jwtBuilder.getEncodedJwt()}`,
+              },
+              body: JSON.stringify({
+                state: "mockState",
+                sub: "mockSub",
+                client_id: "mockClientId",
+                govuk_signin_journey_id: "mockGovukSigninJourneyId",
+              }),
+            });
+            dependencies.getSessionService = () =>
+              new MockSessionServiceSessionRecovered(
+                env.SESSION_TABLE_NAME,
+                env.SESSION_TABLE_SUBJECT_IDENTIFIER_INDEX_NAME,
+              );
+
+            const result = await lambdaHandler(event, dependencies);
+
+            expect(result).toStrictEqual({
+              headers: { "Content-Type": "application/json" },
+              statusCode: 200,
+              body: JSON.stringify({
+                sub: "mockSub",
+                "https://vocab.account.gov.uk/v1/credentialStatus": "pending",
+              }),
+            });
+          });
         });
       });
     });
 
-    describe("Given service returns success response", () => {
-      describe("Given response value is the authSessionId string", () => {
-        it("Returns 200 session recovered response", async () => {
+    describe("Session creation", () => {
+      describe("Given the service returns an error response", () => {
```

Comment:
Given there is an error creating the session

Reply:
Done